### ISSUE TITLE
434ms - push button - JUDCO 1024-2 (norm, 332nF parallel cap, norm)

### ISSUE THREAD

**Issue by adamfk** [adamfk/bouncy-button-data#17]
Please note: 
- This issue is for a **particular instance** of a switch/button of the original poster. If someone else wants to test the same switch model, they should create a new issue.
- The original poster can add additional test sessions to this issue (if they wish) by adding a comment.

# Experiment
Ran this experiment right after https://github.com/adamfk/bouncy-button-data/issues/16. Connected cap and oscilloscope to grab captures. I think too much use with the cap in parallel was causing damage to the switch. Very long bounces.

# Switch Info
See https://github.com/adamfk/bouncy-button-data/issues/6

<br><br>

# Test Session 1

## Setup

Property                                      | Description
----------------------------------- | -----------------------------------------------------
Arduino model                            | Uno (DIP)
Pull-up resistor                            | Arduino internal (measured 35.7k ohm)
Switch wires connection              | Wire leads
Measure switch's ON resistance | 0.5 ohms
Anything else connected?           | 332nF and oscilloscope

**Test Setup Photos:**
See https://github.com/adamfk/bouncy-button-data/issues/16

<br><br>

### 📄 Raw Data
[332nf.txt](https://github.com/adamfk/bouncy-button-data/files/14810087/332nf.txt)

You can graph/analyze the data [here.](https://adamfk.github.io/bouncy-button/bbb-browser/gui.html)

### 🖼️ Summary Image
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/fb912fe9-47f8-4002-af80-c8b577143b4d)

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/06271c9b-c6a6-4415-8408-cb4d8b16f16f)

### 🕵️ Your Observations (optional)
Continuing on from what we saw at the end of testing https://github.com/adamfk/bouncy-button-data/issues/16, we see release max bounces are 3x that of press max bounces.

Before adding the cap, presses would bounce more.

**NOTE!** Adding a cap, delays the release bounce times by about 10ms (see below image) before MCU notices.
The time constant for the 332nF capacitor to charge up through 35.7k ohm resistor is around 11.85ms.

Discharge time constant calculation: 332nF (DMM measured) x 0.3 ohms (I can't measure very low on my DMM) = 99.6ns.

Below capture shows charging up time constant. Estimate at 11.98 ms for 1 tau. 

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/4f0ae436-59e9-42da-8e30-71d8b2959fb4)

---

<br><br><br>

# Some Oscilloscope Captures

## Zoomed in far:
These were a pretty common response:
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/28eaf1bf-e87a-4fce-9556-4c9ba86a0d8b)

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/9bf29a5c-9b40-4a64-8e81-64d6acfb4e72)

Every few times, we would get a response like this though:

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/cd6c2df9-6781-40e6-9666-3c511cbb408a)

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/2c6daebc-d0f4-4bd1-821c-8e9419038dfc)

## 20 us/div
This one was really cool. It really shows that there was a super brief contact, and then didn't contact again until around 97us.
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/1464d58f-b52b-48f1-8c7d-c3d578e4c79c)

Here's the Arduino recorded data:
```
Press the button (it is currently released == logic high)
<test>
sw version: 0.3.3
<buffered_events>
pin: 1, rising_edges: 0, nsec: 0, log: 1, event: 1
pin: 1, rising_edges: 1, nsec: 687, log: 2, event: 2
pin: 0, rising_edges: 0, nsec: 69812, log: 3, event: 3
</buffered_events>
logs used: 3/350
section type: "Normal"
section count: 66
test id: 66
duration: 69 us, 812 ns
</test>
```
Super interesting to note (and maybe problematic) that Arduino sketch didn't record what we expected. If we go 69us back from large falling edge, we see a small noise spike. Could that have been the initial trigger recorded by the Arduino?

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/8c4a2ebb-53a8-4226-b3d7-b067c5b57fd5)

The initial noise at the moment of press just barely went below the approximate voltage required to read a low signal (2.145 vs approx 2.2v): 
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/3a68e6c4-99b6-4db6-a4e5-a457fcee4edd)

---

The transition noise and ringing was detect on this one:
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/b1ae5621-0c4b-42a5-9052-faea9b5b8cdc)

```
<test>
sw version: 0.3.3
<buffered_events>
pin: 1, rising_edges: 0, nsec: 0, log: 1, event: 1
pin: 0, rising_edges: 2, nsec: 812, log: 2, event: 2
pin: 0, rising_edges: 1, nsec: 2375, log: 3, event: 3
pin: 1, rising_edges: 1, nsec: 11153437, log: 6, event: 4
pin: 0, rising_edges: 0, nsec: 29146750, log: 12, event: 5
</buffered_events>
logs used: 12/350
section type: "Normal"
section count: 68
test id: 68
duration: 29 ms, 146 us, 750 ns
</test>
```

---

Another interesting one:
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/85a1d04a-d941-476d-8f06-509889955a9c)

Arduino missed the first low transition. It triggers on the middle pulse.
```
<test>
sw version: 0.3.3
<buffered_events>
pin: 1, rising_edges: 0, nsec: 0, log: 1, event: 1
pin: 1, rising_edges: 1, nsec: 687, log: 2, event: 2
pin: 1, rising_edges: 2, nsec: 2250, log: 3, event: 3
pin: 1, rising_edges: 1, nsec: 19125, log: 4, event: 4
pin: 1, rising_edges: 1, nsec: 32437, log: 5, event: 5
pin: 0, rising_edges: 1, nsec: 40875, log: 6, event: 6
pin: 1, rising_edges: 1, nsec: 43375, log: 7, event: 7
pin: 0, rising_edges: 3, nsec: 45875, log: 8, event: 8
pin: 0, rising_edges: 1, nsec: 137562, log: 9, event: 9
</buffered_events>
logs used: 9/350
section type: "Normal"
section count: 76
test id: 76
duration: 137 us, 562 ns
</test>
```

![image](https://github.com/adamfk/bouncy-button-data/assets/274012/0fb9de53-18b3-44d0-beff-a0b7cedbbe4d)

Those all line up well except for the 19us. There's no trace of anything on the scope that explains that. Note that it was picked up by the rising edge hardware counter. The picoscope was sampling every 40ns (25 MS/s). It is only a 10 MHz scope so it is possible that it missed it.

---

This one the Arduino captured perfectly:
![image](https://github.com/adamfk/bouncy-button-data/assets/274012/c45c1a42-77f1-4fcb-b15d-ddbcbe2f586e)

```
<test>
sw version: 0.3.3
<buffered_events>
pin: 1, rising_edges: 0, nsec: 0, log: 1, event: 1
pin: 1, rising_edges: 1, nsec: 687, log: 2, event: 2
pin: 1, rising_edges: 1, nsec: 168500, log: 3, event: 3
pin: 0, rising_edges: 1, nsec: 467875, log: 4, event: 4
</buffered_events>
logs used: 4/350
section type: "Normal"
section count: 115
test id: 115
duration: 467 us, 875 ns
</test>
```
